### ISSUE TITLE
Click on foreign key yields error: <e.replaceAll is not a function>

### ISSUE THREAD

**Issue by pythoniac** [beekeeper-studio/beekeeper-studio#1580]
**Describe the bug**
Click on a foreign key opens a new tab with the referenced table, filter bar goes to field <ID> and <equals> and <value> ist filled with the correct ID of the foreign key. Then <Error loading data, see below> is displayed and the message in the lowermost edge of the window displays <e.replaceAll is not a function>

**To Reproduce**
Open a table with foreign keys and click on a key.

**Expected behavior**
Open new tab with the table containing the data, filter for foreign key and display the data.

**Screenshots / Gifs**
![err](https://user-images.githubusercontent.com/22104785/232761516-f65dfb55-d336-4dfe-b644-65a6a95ce6e4.gif)

**Version Information (please complete the following information):**
 - OS: Windows 10 v 22H2 (19045.2846)
 - App Version 3.9.8
 - MS-SQL

**Additional context**
Bug exists since initial release of Beekeeper Studio :-(

**Comment by rathboma** [comment]
Hi! We don't see this for most people, so there's probably something unique to your situation that we can't replicate.

Could you give me some SQL to create tables & insert data that can replicate this problem?

**Comment by rathboma** [comment]
Also can I ask -- what version of Sql Server are you using?

**Comment by pythoniac** [comment]
> Hi, I'm on a rather old SQL Server 11.0.7507.2 :-(

**Comment by pythoniac** [comment]
> Hi! We don't see this for most people, so there's probably something unique to your situation that we can't replicate.
> 
> Could you give me some SQL to create tables & insert data that can replicate this problem?

Minimal working example:

--- ORIGINAL TABLE ---

CREATE TABLE [dbo].[00_Car_Brands](
	[ID] [int] IDENTITY(1,1) NOT NULL,
	[Brand] [nvarchar](50) NULL,
 CONSTRAINT [PK_00_Car_Brands] PRIMARY KEY CLUSTERED 
(
	[ID] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO

--- TABLE CONTAINING THE FOREIGN KEY ---

CREATE TABLE [dbo].[00_Car_Details](
	[ID] [int] IDENTITY(1,1) NOT NULL,
	[Brand_ID] [int] NOT NULL,
	[Country] [nvarchar](50) NULL,
 CONSTRAINT [PK_00_Car_Details] PRIMARY KEY CLUSTERED 
(
	[ID] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO

ALTER TABLE [dbo].[00_Car_Details]  WITH CHECK ADD  CONSTRAINT [FK_00_Car_Details_00_Car_Brands] FOREIGN KEY([Brand_ID])
REFERENCES [dbo].[00_Car_Brands] ([ID])
GO

ALTER TABLE [dbo].[00_Car_Details] CHECK CONSTRAINT [FK_00_Car_Details_00_Car_Brands]
GO

**Comment by randyclassen** [comment]
I'm also getting this error. Here's a partial log if that helps at all.

[1683810695381.log](https://github.com/beekeeper-studio/beekeeper-studio/files/11452589/1683810695381.log)

```
SELECT @@version
---
Microsoft SQL Server 2019 (RTM-GDR) (KB5021125) - 15.0.2101.7 (X64) 
	Jan 23 2023 13:08:05 
	Copyright (C) 2019 Microsoft Corporation
	Express Edition (64-bit) on Windows 10 Pro 10.0 <X64> (Build 19045: )
```

**Comment by pythoniac** [comment]
Hello. Unfortunately nothing seems to happen. Maybe itg helps to point out that this issue and < BUG: SQL Server opening a related table via column is throwing an exception #1523 > are the exact same problem. Within the accepted bug report <create table> statements were requested - I have given the statements in question within this thread. So everything should be in place to resolve the issue, right? Thanks a lot!

**Comment by pythoniac** [comment]
Resolved as of version 4.0.3

**Comment by pythoniac** [comment]
Resolved as of version 4.0.3
